File: factory/src/main/java/com/zuzex/factory/mapper/OrderMapper.java
package com.zuzex.factory.mapper;

import com.zuzex.common.dto.CarStatusDto;
import com.zuzex.common.dto.OrderDto;
import com.zuzex.common.grpc.dto.CommonDto;
import com.zuzex.common.model.EventStatus;
import com.zuzex.common.model.Status;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface OrderMapper {

    CarStatusDto toCarStatusDto(Long orderId, Long carId, Status status);

    OrderDto grpcOrderDtoToOrderDto(CommonDto.OrderDto orderDto);

    CommonDto.OrderResultDto toGrpcOrderResultDto(Long carId, EventStatus eventStatus);
}

File: car-factory/src/main/kotlin/com/zuzex/carfactory/mapper/OrderMapper.kt
package com.zuzex.carfactory.mapper

import com.zuzex.common.dto.OrderDto
import com.zuzex.common.grpc.dto.CommonDto
import com.zuzex.common.model.EventStatus
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface OrderMapper {

    fun grpcOrderDtoToOrderDto(orderDto: CommonDto.OrderDto): OrderDto

    fun toGrpcOrderResultDto(carId: Long?, eventStatus: EventStatus): CommonDto.OrderResultDto
}
